src/gui/sdlmain.cpp

```diff
@@ -675,38 +712,59 @@ Bitu GFX_SetSize(Bitu width,Bitu height,Bitu flags,double scalex,double scaley,G
 			free(sdl.opengl.framebuf);
 		}
 		sdl.opengl.framebuf=0;
-		if (!(flags&GFX_CAN_32)) goto dosurface;
-		int texsize=2 << int_log2(width > height ? width : height);
+		if (!(flags & GFX_CAN_32))
+			goto dosurface;
+		int texsize = 2 << int_log2(width > height ? width : height);
 		if (texsize>sdl.opengl.max_texsize) {
 			LOG_MSG("SDL:OPENGL: No support for texturesize of %d, falling back to surface",texsize);
 			goto dosurface;
 		}
-		SDL_GL_SetAttribute( SDL_GL_DOUBLEBUFFER, 1 );
-#if SDL_VERSION_ATLEAST(1, 2, 11)
-		SDL_GL_SetAttribute( SDL_GL_SWAP_CONTROL, 0 );
-#endif
-		GFX_SetupSurfaceScaled(SDL_OPENGL,0);
-		if (!sdl.surface || sdl.surface->format->BitsPerPixel<15) {
-			LOG_MSG("SDL:OPENGL: Can't open drawing surface, are you running in 16bpp (or higher) mode?");
+		SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
```

Comment:
On the Raspberry Pi, disabling double buffering has reduced CPU usage and improved latency with DOSBox (as well as retroarch emulators).  Noticed we got rid of that boolean in the `.conf`, but it would be nice to pass that value through here and let it be configurable.  Default can be enabled.

Reply:
Do you mean disabling it with SDL1 implementation or with SDL2 and just replacing this line with `SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);`?

I agree, that disabling double-buffering should be an option again. However I'm a bit wary of just flicking this value to `0` here, as `SDL_GL_SwapWindow` is used elsewhere in the code (unconditionally). I need to do some more investigation here…

Reply:
Hmm, I found one StackOverflow answer suggesting, that double buffering can't be disabled in SDL2 when using hardware accelerated rendering… https://stackoverflow.com/questions/28341281/how-can-i-do-double-buffering-in-sdl-2-0 I haven't found anything specific in SDL documentation yet, though.